### DISCUSSION TITLE
How can i get regular I2P console to i2Pd on Linux Debian?

### DISCUSSION THREAD

**Discussion by slrslr** [PurpleI2P/i2pd#2064] [Q&A]
Hello, on Debian, I have installed I2Pd package and having access to a working webconsole at port :7070
Yet, i would like to go beyond and test other I2P features, possibly using console like "regular" I2P offers, which allows more graphical interaction to a newbie. How to install it? Possibly i do not need it 24/7 running to eat my CPU/RAM resources. Just to be there in case i want to run it and do something in it. Thank you

**Comment by anikey-from-i2p** [comment]
If i understand correctly, you want to use the webconsole of Java I2P while using i2pd instead of Java i2p?

In that case, that is not possible. Webconsoles are more or less built into the respective routers.

If you want to use Java i2p's console, you use java i2p. If you want i2pd webconsole, then use i2pd.

In i2pd many things are done through config files instead of webconsole. Don't fear it, it's not hard. Just read the docs ([this](https://i2pd.readthedocs.io/en/latest/user-guide/configuration/) and [this](https://i2pd.readthedocs.io/en/latest/user-guide/tunnels/), also be sure to check other pages on that site). And don't forget that config file changes are only applied either when you restart i2pd or you tell it to reload the config files (either by a posix signal or via the webconsole).

<details><summary>advanced users only (click to show)</summary>
<p>

Well actually there appears to exist a standard i2p protocol called "I2PControl" (spec [inside i2p](http://i2p-projekt.i2p/en/docs/api/i2pcontrol) and [on clearnet](https://geti2p.net/en/docs/api/i2pcontrol)) which seems to allow to control i2p router through a json interface.

However i haven't seen it actually used anywhere. (and i've put this text under a hidden section exactly because of that). But i guess someone could in theory take that protocol and make a unified control GUI for all routers.

</p>
</details>
